code/modules/modular_armor/attachments.dm

```diff
@@ -0,0 +1,687 @@
+	icon_state = "assault_arms"
+	greyscale_config = /datum/greyscale_config/modulararms_assault
+
+/obj/item/armor_module/armor/arms/marine/eva
+	name = "\improper Jaeger Pattern EVA arm plates"
+	desc = "Designed for use with the Jaeger Combat Exoskeleton. It provides protection and encumbrance when attached and is fairly easy to attach and remove from armor. Click on the armor frame to attach it. This armor appears to be marked as a EVA armor piece."
+	icon_state = "eva_arms"
+	greyscale_config = /datum/greyscale_config/modulararms_eva
+
+/obj/item/armor_module/armor/arms/marine/eod
+	name = "\improper Jaeger Pattern EOD arm plates"
+	desc = "Designed for use with the Jaeger Combat Exoskeleton. It provides protection and encumbrance when attached and is fairly easy to attach and remove from armor. Click on the armor frame to attach it. This armor appears to be marked as a EOD armor piece."
+	icon_state = "eod_arms"
+	greyscale_config = /datum/greyscale_config/modulararms_eod
+
+
+/*
+	Visors
+*/
+
+/obj/item/armor_module/armor/visor
+	name = "standard visor"
+	slot = ATTACHMENT_SLOT_VISOR
+	///Initial hex color we use when applying the visor color
+	greyscale_colors = "#f7fb58"
+	flags_attach_features = ATTACH_SAME_ICON
+	///whether this helmet should be using its emissive overlay or not
+	var/visor_emissive_on = TRUE
+	///Whether or not the helmet is allowed to turn its emissive on or off.
+	var/allow_emissive = TRUE
+
+
+/obj/item/armor_module/armor/visor/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	to_chat(user, "Click the visor with paint to color the visor.")
+
+/obj/item/armor_module/armor/visor/on_attach(obj/item/attaching_to, mob/user)
+	. = ..()
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY_ALTERNATE, .proc/handle_color)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/extra_examine)
+	RegisterSignal(parent, COMSIG_CLICK_RIGHT, .proc/toggle_emissive)
+	if(visor_emissive_on)
+		parent.AddElement(/datum/element/special_clothing_overlay/modular_helmet_visor, HEAD_LAYER, greyscale_config, greyscale_colors)
+	update_icon()
+
+/obj/item/armor_module/armor/visor/on_detach(obj/item/detaching_from, mob/user)
+	UnregisterSignal(parent, list(COMSIG_CLICK_RIGHT, COMSIG_PARENT_EXAMINE, COMSIG_PARENT_ATTACKBY_ALTERNATE))
+	parent.RemoveElement(/datum/element/special_clothing_overlay/modular_helmet_visor, HEAD_LAYER, greyscale_config, greyscale_colors)
+	update_icon()
+	return ..()
+
+///Colors the visor when the parent is right clicked with facepaint.
+/obj/item/armor_module/armor/visor/proc/handle_color(datum/source, obj/I, mob/user)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, /atom/proc/attackby, I, user)
+	return COMPONENT_NO_AFTERATTACK
+
+///Toggles the visors emmisiveness if allowed.
+/obj/item/armor_module/armor/visor/proc/toggle_emissive(datum/source, mob/user)
+	SIGNAL_HANDLER
+	if(!greyscale_config || !allow_emissive)
+		return
```

Comment:
for the sake of organization please use multiple files here
ie helmtet/chestölate0/moduless for example